enterprise/backup/src/main/java/org/neo4j/backup/impl/OnlineBackupCommandBuilder.java

```diff
@@ -161,13 +168,20 @@ public boolean backup( String targetLocation ) throws CommandFailed, IncorrectUs
     public boolean backup( File targetLocation ) throws CommandFailed, IncorrectUsage
     {
         String[] args;
-        try
+        if ( rawArgs.isPresent() )
```

Comment:
See above. Check for empty array

Reply:
The thing is, an empty array is valid and has a different meaning than null. The Optional forces the implementation to acknowledge that a) this is nullable and b) that in the event of an empty array, the empty array takes precedence over other flags manually set with the builder.
So if you create the builder with an address, location and additional config, if you provide raw arguments that will override the previously set options.